csharp/LogicalWrite.cs

```diff
@@ -479,27 +514,104 @@ private static void ConvertByteArray(ReadOnlySpan<byte[]> source, Span<short> de
                 }
                 else
                 {
-                    destination[dst++] = FromByteArray(value, byteBuffer);
+                    destination[dst++] = LogicalWrite.FromByteArray(value, byteBuffer);
                     defLevels[i] = (short) (nullLevel + 1);
                 }
             }
         }
 
-        private static unsafe ByteArray FromString(string str, ByteBuffer byteBuffer)
+
+    }
+
+    /// <summary>
+    /// C# types to Parquet physical types write conversion logic.
+    /// Separate class for per-element conversion logic.
+    /// </summary>
+    internal static class LogicalWrite
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static FixedLenByteArray FromDecimal(decimal source, decimal multiplier, ByteBuffer byteBuffer)
+        {
+            var dec = new Decimal128(source, multiplier);
+            return FromFixedLength(in dec, byteBuffer);
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static unsafe FixedLenByteArray FromUuid(Guid uuid, ByteBuffer byteBuffer)
+        {
+            Debug.Assert(sizeof(Guid) == 16);
+
+            // The creation of a temporary byte[] via ToByteArray() is a shame, but I can't find a better public interface into Guid.
+            // Riskier but faster proposition is to assume that the layout of Guid is consistent. There is no such guarantees!
+            // But hopefully any breaking change is going to be caught by our unit test.
+
+            //var array = FromFixedLengthByteArray(uuid.ToByteArray(), byteBuffer); // SLOW
+            var array = FromFixedLength(uuid, byteBuffer);
+            var p = (byte*) array.Pointer;
+
+            // From parquet-format logical type documentation
+            // The value is encoded using big-endian, so that 00112233-4455-6677-8899-aabbccddeeff is encoded
+            // as the bytes 00 11 22 33 44 55 66 77 88 99 aa bb cc dd ee ff.
+            //
+            // But Guid (and even ToByteArray()) uses a little endian representation.
```

Comment:
dword,word,word,byte[8]. First 3 parts platform-endianness.
Q: are you wrong on <obscure big endian platform>?